src/source/core/ComponentPool.bs

```diff
@@ -60,10 +60,11 @@ namespace mc
       compName = parts[0]
       pool = m.getComponentPool(componentType)
 
-      for i = 0 to number
+      for i = 0 to number - 1
         component = createObject("roSGNode", compName)
         if component <> invalid
           component.id = rnd(2147483647).toStr()
+          mc.setOrAddNodeField(component, "componentType", componentType)
```

Comment:
@dacianroman can you please name this _mcomponentType - so it won't collide with any other code.. we've no idea who will use this in future :)

Reply:
updated to _mComponentType